[2025-04-15T13:00:32.817+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: enrich_staging_rows.run_enrich_staging_rows manual__2025-04-15T13:00:29.863785+00:00 [queued]>
[2025-04-15T13:00:32.823+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: enrich_staging_rows.run_enrich_staging_rows manual__2025-04-15T13:00:29.863785+00:00 [queued]>
[2025-04-15T13:00:32.824+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 2
[2025-04-15T13:00:32.834+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): run_enrich_staging_rows> on 2025-04-15 13:00:29.863785+00:00
[2025-04-15T13:00:32.844+0000] {standard_task_runner.py:60} INFO - Started process 18554 to run task
[2025-04-15T13:00:32.849+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'enrich_staging_rows', 'run_enrich_staging_rows', 'manual__2025-04-15T13:00:29.863785+00:00', '--job-id', '229', '--raw', '--subdir', 'DAGS_FOLDER/enrich_staging_rows_dag.py', '--cfg-path', '/tmp/tmpcr0yehf7']
[2025-04-15T13:00:32.856+0000] {standard_task_runner.py:88} INFO - Job 229: Subtask run_enrich_staging_rows
[2025-04-15T13:00:32.914+0000] {task_command.py:423} INFO - Running <TaskInstance: enrich_staging_rows.run_enrich_staging_rows manual__2025-04-15T13:00:29.863785+00:00 [running]> on host d8146153ddc3
[2025-04-15T13:00:32.966+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='enrich_staging_rows' AIRFLOW_CTX_TASK_ID='run_enrich_staging_rows' AIRFLOW_CTX_EXECUTION_DATE='2025-04-15T13:00:29.863785+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-15T13:00:29.863785+00:00'
[2025-04-15T13:00:32.969+0000] {connection.py:370} INFO - Snowflake Connector for Python Version: 3.6.0, Python Version: 3.8.18, Platform: Linux-6.12.5-linuxkit-aarch64-with-glibc2.34
[2025-04-15T13:00:32.970+0000] {connection.py:1171} INFO - This connection is in OCSP Fail Open Mode. TLS Certificates would be checked for validity and revocation status. Any other Certificate Revocation related exceptions or OCSP Responder failures would be disregarded in favor of connectivity.
[2025-04-15T13:00:33.609+0000] {cursor.py:1028} INFO - query: [SELECT * FROM SAMPLED_SONGS_STAGING WHERE artist IS NULL OR genre IS NULL]
[2025-04-15T13:00:34.001+0000] {cursor.py:1041} INFO - query execution done
[2025-04-15T13:00:34.004+0000] {cursor.py:1205} INFO - Number of results in first chunk: 45
[2025-04-15T13:00:34.020+0000] {logging_mixin.py:188} INFO - ⚠️ Failed to parse LLM response: name 'title' is not defined
[2025-04-15T13:00:34.020+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/opt/airflow/backend/rag_pipeline/enrich_staging_rows.py", line 53, in enrich_staging_rows
    Title: {title}
NameError: name 'title' is not defined

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/backend/rag_pipeline/enrich_staging_rows.py", line 91, in enrich_staging_rows
    print("Response content:", raw)
UnboundLocalError: local variable 'raw' referenced before assignment
[2025-04-15T13:00:34.040+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=enrich_staging_rows, task_id=run_enrich_staging_rows, execution_date=20250415T130029, start_date=20250415T130032, end_date=20250415T130034
[2025-04-15T13:00:34.048+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 229 for task run_enrich_staging_rows (local variable 'raw' referenced before assignment; 18554)
[2025-04-15T13:00:34.076+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-04-15T13:00:34.104+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
